src/ig/IGExporter.ts

```diff
@@ -1338,96 +1342,169 @@ export class IGExporter {
+        parameter.code.system = 'http://hl7.org/fhir/tools/CodeSystem/ig-parameters';
+      }
+    });
+
+    // Add new copyrightLabel property
+    if (this.config.copyrightLabel) {
+      this.ig.copyrightLabel = this.config.copyrightLabel;
+    }
 
-      // Add new copyrightLabel property
-      if (this.config.copyrightLabel) {
-        this.ig.copyrightLabel = this.config.copyrightLabel;
+    // Add new versionAlgorithm property
+    if (this.config.versionAlgorithmString) {
+      this.ig.versionAlgorithmString = this.config.versionAlgorithmString;
+    } else if (this.config.versionAlgorithmCoding) {
+      this.ig.versionAlgorithmCoding = this.config.versionAlgorithmCoding;
+    }
+
+    // Add new dependsOn.reason property
+    this.ig.dependsOn?.forEach(dependency => {
+      const configDependency = this.config.dependencies?.find(
+        d => d.packageId === dependency.packageId
+      );
+      if (configDependency.reason) {
+        dependency.reason = configDependency.reason;
       }
+    });
+  }
 
-      // Add new versionAlgorithm property
-      if (this.config.versionAlgorithmString) {
-        this.ig.versionAlgorithmString = this.config.versionAlgorithmString;
-      } else if (this.config.versionAlgorithmCoding) {
-        this.ig.versionAlgorithmCoding = this.config.versionAlgorithmCoding;
+  // If an R4 IG uses any new properties from R5, they should be included as extensions.
+  // Converting between FHIR versions is documented here: http://hl7.org/fhir/2022Sep/versions.html#extensions
+  addR5ExtensionsToR4() {
+    // Any new properties added to the IG resource in R5 should be included on extensions in R4 IGs
+    this.ig.definition.resource.forEach(resource => {
+      // Assign IG.definition.resource.profile to an extensions if provided
+      const configEntry = this.config.resources?.find(
+        r => r.reference?.reference === resource.reference?.reference
+      );
+      if (configEntry?.profile != null) {
+        resource.extension = [
+          ...(resource.extension ?? []),
+          {
+            url: 'http://hl7.org/fhir/5.0/StructureDefinition/extension-ImplementationGuide.definition.resource.profile',
+            valueCanonical: configEntry.profile
+          }
+        ];
       }
+    });
 
-      // Add new dependsOn.reason property
-      this.ig.dependsOn?.forEach(dependency => {
-        const configDependency = this.config.dependencies?.find(
-          d => d.packageId === dependency.packageId
-        );
-        if (configDependency.reason) {
-          dependency.reason = configDependency.reason;
+    // Add new IG.definition.page.source[x] property to an extension if it is provided. No need to set a default like R5 needs to.
+    this.ig.definition.page.page.forEach(page => {
+      // this.ig.definition.page is the toc.html page we create
+      // All configured pages are at the next level, so start at that level
```

Comment:
This confused me at first, until I realized you were really commenting on why you iterate on `this.ig.definition.page.page` in the line _above_.  Should this comment maybe be above the line it is explaining?

Reply:
I think that's fair. I will confess that I think I copied that from the previous R5 functionality, so I'll update it in both places.

Reply:
Comments moved in [f396edf](https://github.com/FHIR/sushi/pull/1189/commits/f396edfe2a6e15631a25977b29463882003c8384).